cpp/server/ct-server.cc

```diff
@@ -174,11 +176,15 @@ void SignMerkleTree(TreeSigner<LoggedCertificate>* tree_signer,
 
   while (true) {
     if (election->IsMaster()) {
-      CHECK_EQ(tree_signer->UpdateTree(), TreeSigner<LoggedCertificate>::OK);
-    } else {
-      VLOG(1) << "Not starting signing run since we're not the master.";
+      util::Status status(tree_signer->SequenceNewEntries());
+      if (!status.ok()) {
+        // TODO(alcutter): Probably don't need to blow up here
+        LOG(FATAL) << "Problem sequencing new entries: " << status;
+      }
```

Comment:
This whole thing looks an awful lot like you wanted this single liner:

    CHECK_EQ(util::Status::OK, tree_signer->SequenceNewEntries()) << "problem sequencing new entries";

:wink: